rct229/rulesets/ashrae9012019/section21/section21rule8.py

```diff
@@ -0,0 +1,176 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.ruleset_model_factory import produce_ruleset_model_instance
+from rct229.rulesets.ashrae9012019 import BASELINE_0
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_hw_loop_zone_list_w_area_dict import (
+    get_hw_loop_zone_list_w_area,
+)
+from rct229.schema.config import ureg
+from rct229.schema.schema_enums import SchemaEnums
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.pint_utils import CalcQ
+from rct229.utils.std_comparisons import std_equal
+
+DESIGN_TEMP_RESET_TYPE = SchemaEnums.schema_enums["TemperatureResetOptions"]
+DESIGN_SUPPLY_TEMP_AT_OUTDOOR_HIGH = 150 * ureg("degF")
+DESIGN_SUPPLY_TEMP_AT_OUTDOOR_LOW = 180 * ureg("degF")
+DESIGN_OUTDOOR_TEMP_FOR_RESET_HIGH = 50 * ureg("degF")
+DESIGN_OUTDOOR_TEMP_FOR_RESET_LOW = 20 * ureg("degF")
+
+
+class Section21Rule8(RuleDefinitionListIndexedBase):
+    """Rule 8 of ASHRAE 90.1-2019 Appendix G Section 21 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section21Rule8, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=False, BASELINE_0=True, PROPOSED=False
+            ),
+            each_rule=Section21Rule8.HeatingFluidLoopRule(),
+            index_rmr=BASELINE_0,
+            id="21-8",
+            description="When the baseline building requires boilers, (for baseline system type = 1,5,7,11 and 12), HWST for the baseline building shall be reset using an outdoor air dry-bulb reset schedule. 180F at 20F OAT, 150Fat 50F OAT, ramped linerarly between 150F and 180F.",
+            ruleset_section_title="HVAC - Water Side",
+            standard_section="Section G3.1.3.3 Building System-Specific Modeling Requirements for the Baseline model",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.fluid_loops[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.BASELINE_0
+        hw_loop_zone_list_w_area_dict_b = get_hw_loop_zone_list_w_area(rmi_b)
+        boiler_loop_ids_b = find_all("boilers[*].loop", rmi_b)
```

Comment:
Could we add ```$``` in the json path as shown below? 

```python
boiler_loop_ids = find_all("$.boilers[*].loop", rmi_b)
```

This works without ```$``` because there is a function that adds ```$```. However, we'll likely remove the function and it's recommended to have a full path (including $).

Reply:
Revised! Thank you for the clarification:)